cloudevents/working-drafts/minor-versions.md

```diff
@@ -0,0 +1,337 @@
+
+It may sound natural to allow something like:
+
+```csharp
+var evt = new CloudEvent { Id = "..." };
+```
+
+... to just use the latest spec version that the SDK supports. However, this is a breaking change.
+Consider:
+
+```csharp
+var evt = new CloudEvent
+{
+    Id = "...",
+    ["xyz"] = "test"
+};
+```
+
+When the SDK only supports v1.0, that creates a valid event. When the SDK supports v1.1 as well,
+that would produce an invalid event or throw an exception, if the version is implicitly v1.1.
+Instead, the SDK should *implicitly* use the latest version that was present when the first stable
+version of the SDK came out, within its own current major version. In other words, if v1.1 ships
+and a new SDK is created, that can default to v1.1 - and if an SDK creates its own new major version,
+*that* can default to v1.1 as well... but in both cases that default should stay as v1.1 in new minor
+SDK versions even after v1.2 ships.
+
+An SDK *may* choose to provide a way of explicitly opting into the "latest supported" version:
+
+```csharp
+var evt = new CloudEvent(CloudEventsSpecVersion.Latest)
+{
+  
+};
+```
+
+### Accessing attributes
+
+If an SDK exposes core attributes in a distinct way from extension attributes (as shown above,
+where `Id` is a strongly-typed property, whereas `["xyz"]` uses an indexer with a string literal)
+then any attempt to use the "special" access should fail when accessing an attribute within
+an event with a version that doesn't have that attribute as a core attribute. Example:
+
+```csharp
+// Create a v1.0 event
+var evt = new CloudEvent(CloudEventsSpecVersion.V1_0);
+
+// Try to access the new Xyz property, referring to the optional attribute introduced in
+// v1.1: this should fail.
+evt.Xyz = 1;
+
+// Access the xyz attribute by name, as an extension attribute.
+// This should work, even though the value wouldn't be valid in a v1.1 event.
+evt["xyz"] = "test";
+
+// Likewise accessing the property for an event parsed from elsewhere:
+evt = httpRequest.ParseCloudEvent();
+// If evt is v1.0, this should fail *even if it has an xyz attribute value that's valid in v1.1.
+int xyz = evt.Xyz;
+```
+
+Setting a value within an event where the version indicates that the attribute *is* known should
+validate the attribute value according to that version. Example:
+
+```csharp
+// Create a v1.1 event
+var evt = new CloudEvent(CloudEventsSpecVersion.V1_1);
+
+// This should work: 1 is a valid value, and xyz is known in v1.1
+evt.Xyz = 1;
+
+// This should fail: 123 is not a valid value for xyz in v1.1
+evt.Xyz = 123;
+
+// This should fail to compile, where that's feasible: xyz should always be an integer
+evt.Xyz = "test";
+
+// This should succeed at execution time
+evt["xyz"] = 123;
+
+// This should fail at execution time due to validation
+evt["xyz"] = "test";
+
+// This should fail at execution time due to validation
+evt["xyz"] = 123;
+```
+
+### "Changing" event versions
```

Comment:
For ref, the go sdk does this: https://github.com/cloudevents/sdk-go/blob/main/v2/event/eventcontext_v1.go#L145 

Each version implementation had a method added that allowed it to move one version up or one version down. You could also implement this in a different way if you had a generic event wrapper with something like a `core` and `extensions` inner members that had versioned implementations. Then you could write a `Upgrade` and `Downgrade` methods that would do the lookup work and/or fail if the event is at the highest/lowest supported versions.